tonytanaka/index.js

```diff
@@ -0,0 +1,75 @@
+// 1
+let board = [
+    [1, 2, 3],
+    ["quick", "brown", "fox", "jumped", "over", "lazy", "dog"],
+    [true, false]
+  ];
+
+function loopArrays (arr) {
+    for (let i of arr) {
+//        console.log(i) // i is each one of the 3 arrays
+ //       console.log(i.length) // 3, 7 and 2
+        for (j in i) {
+            console.log(i[j])
+        }
+    }
+}
+console.log(loopArrays(board)) 
+/*
+ 1
+2
+3
+quick
+brown
+fox
+jumped
+over
+lazy
+dog
+true
+false
+==> undefined <== !!!
+*/
+//  why am I getting and UNDEFINED at the end of the list?
+
+
+// 2
+// 2.1 Create a doggo object. Add name and breed as properties of the object.
+// 2.2 Create an array within the doggo object named favorite foods and add the doggo's favorite foods to the array.
+const doggo = {
+    name: "Kelly",
+    breed: "Lhasa Apso",
+    favoriteFoods: ["fruits", "chicken", "sausage"]
+}
+
+// 2.3 Access the second element of the doggo's favorite foods. 
+for (let key in doggo) {
+    if (key === 'favoriteFoods') {
+        console.log("The 2nd fav food is:", doggo.favoriteFoods[1])
+    }
+}
+
+// 2.4 Add a method that loops through and print all the doggo's favorite food.
```

Comment:
Task 2.4 ==> You can declare a method within doggo object like for example after favoriteFoods array
favFood() {
this.favoriteFood.map((food) => console.log("FavFood==>",food));
}
And then after doggo object you can call the function and print the favoriteFood using doggo.favFood()

Reply:
tks Ciamac!
please take a look also at the first exercise, it comes an "undefined" result at the end, and I don't know why
the same happens to this new code you taught me